packages/ui-react/src/components/Checkbox/Checkbox.tsx

```diff
@@ -1,60 +1,44 @@
-import React, { type ReactNode } from 'react';
+import React, { type InputHTMLAttributes } from 'react';
 import classNames from 'classnames';
-import { useTranslation } from 'react-i18next';
-import useOpaqueId from '@jwp/ott-hooks-react/src/useOpaqueId';
 
-import HelperText from '../HelperText/HelperText';
+import { FormField } from '../FormField/FormField';
+import type { FieldProps } from '../../types/form-fields';
 
 import styles from './Checkbox.module.scss';
 
-type Props = {
-  label?: ReactNode;
-  name: string;
-  value?: string;
+type HTMLCheckboxProps = Omit<InputHTMLAttributes<HTMLInputElement>, 'size'>;
+
+type Props = HTMLCheckboxProps & {
   checked?: boolean;
   onChange: React.ChangeEventHandler<HTMLInputElement>;
   header?: string;
-  error?: boolean;
-  helperText?: string;
-  disabled?: boolean;
-  required?: boolean;
-  lang?: string;
-};
+  helperText?: React.ReactNode;
+} & FieldProps;
```

Comment:
Does it make sense to have a separate `fieldProps` prop?

Reply:
I'm not sure. The Checkbox component already is a field component, so I don't think having `fieldProps` as a prop makes sense. However, I'm familiar with patterns passing props as named props to the input like so:

```ts
const Checkbox = ({ inputProps, label, required, ...rest }) => (
  return (
    <FormField 
      label={label} 
      input={<Input type="checkbox" {...inputProps} required={required} />}
    />
  );
);
```

Reply:
I'm trying this, but I'm running into some challenges... I like having the `inputProps` categorized, but then we have to decide if we;

- borrow props from `inputProps` to fill in required `FormField` props
- have some duplicated props (name, required, editing, etc) that are used in most Inputs and the FormField component.

For example;

```tsx
type HTMLCheckboxProps = Omit<InputHTMLAttributes<HTMLInputElement>, 'size'>;

// imported from Input control component?
type InputProps = HTMLCheckboxProps & {
  label: React.ReactNode;
  editing?: boolean; // replace with `readOnly`?
};

type Props = {
  inputProps: InputProps;
} & FormFieldProps;

const Checkbox = ({ inputProps, ...fieldProps }) => {
  return (
    <FormField
      name={inputProps.name}
      required={inputProps.required}
      editing={inputProps.editing}
      renderInput={() => <input type="checkbox" {...inputProps} />}
      {...fieldProps}
    />
  );
};
```

Reply:
I have tested it like so:

```tsx
export type Props = {
  className?: string;
  helperText?: React.ReactNode;
  label?: React.ReactNode;
  inputProps: InputProps;
  testId?: string;
};

const TextField: React.FC<Props> = ({ className, helperText, label, testId, inputProps = { name: '' } }) => {
  return (
    <FormField
      className={classNames(styles.textField, className)}
      error={inputProps.error}
      helperText={helperText}
      label={label}
      required={inputProps.required}
      editing={inputProps.editing}
      testId={testId}
      name={inputProps.name}
      renderInput={(fieldProps) => <Input {...fieldProps} {...inputProps} />}
    />
  );
};
```

This does make the field components a lot cleaner (props and JSX), but it does mean every usage will change (which I tried to prevent)

The usage looks like:

```tsx
const renderInputs = () => (
  <>
    <TextField label="Default" inputProps={{ value: '', name: 'input', placeholder: 'placeholder' }} />
    <TextField label="Required" inputProps={{ value: '', name: 'input', placeholder: 'placeholder', required: true }} />
    <TextField label="Disabled" inputProps={{ value: '', name: 'input', placeholder: 'placeholder', disabled: true }} />
    <TextField label="Required with error" inputProps={{ value: '', name: 'input', placeholder: 'placeholder', required: true, error: true }} />
    <TextField label="Helper text" helperText="This field is required" inputProps={{ value: '', name: 'input', placeholder: 'placeholder' }} />
    <TextField
      label="With right control"
      inputProps={{
        value: '',
        name: 'input',
        placeholder: 'placeholder',
        required: true,
        error: true,
        rightControl: (
          <IconButton>
            <Icon icon={Visibility} />
          </IconButton>
        ),
      }}
    />
    <TextField
      label="With left control"
      inputProps={{
        value: '',
        name: 'input',
        placeholder: 'placeholder',
        required: true,
        error: true,
        leftControl: (
          <IconButton>
            <Icon icon={Visibility} />
          </IconButton>
        ),
      }}
    />
    <TextField label="Read only" inputProps={{ value: '', name: 'input', placeholder: 'placeholder', editing: false }} />
  </>
);
```

What do you think? 

Ping @AntonLantukh @dbudzins @royschut @langemike @MelissaDTH

Reply:
I like the pattern (much like MUI). 
However I suspect many merge conflicts for work-in-progress branches, so perhaps we should time it smart?

Reply:
In case you expect to have merge conflicts, it is also okay to implement it later.